content/vstools/reading/debugger/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Debugger"
+date: 2023-02-14T19:56:25-06:00
+draft: false
+weight: 3
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Some of the features that you will find yourself leaning on the most as you further your career in tech is the Visual Studio debugging tools.
+
+Open up your [examples repo](https://github.com/LaunchCodeEducation/csharp-web-dev-examples) and find the `HelloMethods` project. Read through each item, make note of how the debugging tools work for each item, and practice using the tool on `HelloMethods`:
+
+1. Right-click in the text editing window to add a [breakpoint](https://learn.microsoft.com/en-us/visualstudio/debugger/debugger-feature-tour?view=vs-2022#set-a-breakpoint-and-start-the-debugger) to your code.
+1. Start debug mode with the run button as you might normally run your program.
+1. Make note of the debugger panes and how you might make use of them:
+   1. [Autos](https://learn.microsoft.com/en-us/visualstudio/debugger/autos-and-locals-windows?view=vs-2022) pane shows the values of parameters and variables on the line that the debugger is 
+     currently sitting on, as well as the line above.
+   1. [Locals](https://learn.microsoft.com/en-us/visualstudio/debugger/autos-and-locals-windows?view=vs-2022) pane shows the value of local variables and parameters within the program being debugged.
+   1. Add variables, parameters, and expressions to the [Watch](https://learn.microsoft.com/en-us/visualstudio/debugger/debugger-feature-tour?view=vs-2022#set-a-watch) pane to monitor their values while debugging.
+   1. [Call Stack](https://learn.microsoft.com/en-us/visualstudio/debugger/how-to-use-the-call-stack-window?view=vs-2022) pane displays the record of the methods that have been called in the program being debugged.
+   - View a list of breakpoints in the *Breakpoints* pane. You may also disable and enable breakpoints from here.
+1. Debugger [Code Stepping](https://learn.microsoft.com/en-us/visualstudio/debugger/navigating-through-code-with-the-debugger?view=vs-2022&tabs=csharp#code-stepping) Buttons:
+   1. [Step over](https://learn.microsoft.com/en-us/visualstudio/debugger/debugger-feature-tour?view=vs-2022#step-over-code-to-skip-functions) button moves debugger to the next line to be executed within a method.
```

Comment:
All these links seem to be for VS, do they also work for VS for Mac?

![image](https://user-images.githubusercontent.com/39868164/220455730-101441bc-49de-4a81-9d7a-e7fc615fe3cf.png)

Reply:
I see there is another section on line 37 -> Extra tips for Mac users.  Maybe make a note around line 16, that this has been include.

Reply:
Done!